### PR TITLE
Implement ord for FixedSizeBinary types

### LINKED ISSUE #2904: Ordering not implemented for FixedSizeBinary types
As the subject says. This is mostly a placeholder for an associated PR.

### PR DESCRIPTION
**Pull request by maxburke**
# Which issue does this PR close?

Closes #2904

# Rationale for this change
 
Implementing sorting / grouping in datafusion on columns of type FixedSizeBinary requires that ord be implemented for FixedSizeBinary types.

### CODE REVIEW

**Review by tustvold — COMMENTED** [review]
Could we get some tests of this please :pray:

File: arrow/src/array/ord.rs [hunk 1]
```diff
@@ -295,6 +295,14 @@ pub fn build_compare(left: &dyn Array, right: &dyn Array) -> Result<DynComparato
             let right: Decimal128Array = Decimal128Array::from(right.data().clone());
             Box::new(move |i, j| left.value(i).cmp(&right.value(j)))
         }
+        (FixedSizeBinary(_), FixedSizeBinary(_)) => {
```
Comment (tustvold) [inline arrow/src/array/ord.rs:298]: FYI the comparison and sort kernels do not make use of this `DynComparator`, and I actually hope to at some point deprecate it in favor of the row format.

**Comment by maxburke** [comment]
Tests added, issue link fixed.

File: arrow/src/array/ord.rs (hunk 1 above)
Comment (maxburke) [reply arrow/src/array/ord.rs:298]: That may be but we were running into the unhandled case in this match firing when testing https://github.com/apache/arrow-datafusion/pull/3911

**Review by tustvold — APPROVED** [review]

File: arrow/src/array/ord.rs [hunk 2]
```diff
@@ -307,10 +315,34 @@ pub fn build_compare(left: &dyn Array, right: &dyn Array) -> Result<DynComparato
 #[cfg(test)]
 pub mod tests {
     use super::*;
-    use crate::array::{Float64Array, Int32Array};
+    use crate::array::{FixedSizeBinaryArray, Float64Array, Int32Array};
     use crate::error::Result;
     use std::cmp::Ordering;
 
+    #[test]
```
Comment (tustvold) [inline arrow/src/array/ord.rs:322]: Even better would be some tests with more than one value :sweat_smile:

**Pull request merged**